src/render_helpers/mod.rs

```diff
@@ -127,7 +132,75 @@ pub fn render_to_dmabuf(
         }
     }
 
-    let _sync_point = frame.finish().context("error finishing frame")?;
+    let sync_point = frame.finish().context("error finishing frame")?;
+
+    Ok(sync_point)
+}
+
+pub fn render_to_shm(
+    renderer: &mut GlesRenderer,
+    buffer: &WlBuffer,
+    region: Rectangle<i32, Physical>,
+    size: Size<i32, Physical>,
```

Comment:
I was thinking, instead of rendering the full `size` and then copying out the `region`, to render already to a small `region.size`-sized texture, by wrapping `elements` in a `RelocateRenderElement` with relative location of `-region.loc`.

Reply:
Done.